src/v/cluster/rm_stm.cc

```diff
@@ -2026,9 +2027,14 @@ void rm_stm::compact_snapshot() {
 
     for (auto it = _log_state.seq_table.cbegin();
          it != _log_state.seq_table.cend();) {
-        if (
-          now.value() - it->second.entry.last_write_timestamp
-          <= expiration_ms_count) {
+        // expired producer must not have any ongoing transactions and be older
+        // than expiration threshold.
+        auto is_expired_producer
+          = now.value() - it->second.entry.last_write_timestamp
+              <= expiration_ms_count
+            && !_log_state.ongoing_map.contains(it->first);
```

Comment:
Each tx has a phase when it's inflight but ongoing_map is empty. When a tx starts it writes `fence_batch` to the partition and only later a client sends first data to the partition which fills `ongoing_map`.

Reply:
got it, will fix it

Reply:
is it enough to check `current_txs` ?